File: java/leetcode_grind/Day849.java
package leetcode_grind;

import java.util.Arrays;

public class Day849 {
    // https://leetcode.com/problems/divide-array-into-equal-pairs/description/?envType=daily-question&envId=2025-03-17
    static class Solution1 {
        public boolean divideArray(int[] nums) {
            Arrays.sort(nums);
            for (int pos = 0; pos < nums.length; pos += 2) {
                if (nums[pos] != nums[pos + 1]) {
                    return false;
                }
            }
            return true;
        }
    }

}

File: src/leetcode_grind/day_849.rs
// https://leetcode.com/problems/divide-array-into-equal-pairs/description/?envType=daily-question&envId=2025-03-17
pub fn divide_array(nums: Vec<i32>) -> bool {
    let max_num = *nums.iter().max().unwrap_or(&0);
    let mut needs_pair = vec![false; max_num as usize + 1];
    for &num in &nums {
        needs_pair[num as usize] = !needs_pair[num as usize];
    }
    for &num in &nums {
        if needs_pair[num as usize] {
            return false;
        }
    }
    true
}